services/QuillCMS/app/controllers/responses_controller.rb

```diff
@@ -40,6 +40,25 @@ def create_or_increment
     render json: {}
   end
 
+  # POST /responses/create_or_update
+  def create_or_update
+    symbolized_vals = transformed_new_vals(params_for_create).to_h.symbolize_keys
+    response = Response.find_by(text: symbolized_vals[:text], question_uid: symbolized_vals[:question_uid])
+    if !response.present?
```

Comment:
It feels like there's some duplication with `create` and `update`.  Could you just refactor lines 47-59 with a rough idea of:

`response.present? ? update : create`

Reply:
Great suggestion. I was able to go back and refactor it a lot more cleanly with `find_or_initialize_by`

Reply:
That looks even better and is more Rails-y.  Great job!